pkg/admiralctl/migrate/migrate.go

```diff
@@ -0,0 +1,694 @@
+	}
+
+	_, err := kubeadmiralClient.CoreV1alpha1().ClusterPropagationPolicies().Get(context.TODO(), policyName, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		cpp := buildClusterPropagationPolicy(policyName, o.Cluster, o.SchedulingMode)
+		_, err = kubeadmiralClient.CoreV1alpha1().ClusterPropagationPolicies().Create(context.TODO(), cpp, metav1.CreateOptions{})
+
+		return policyName, err
+	}
+	if err != nil {
+		return "", fmt.Errorf("failed to get ClusterPropagationPolicy(%s) in control plane: %w", policyName, err)
+	}
+
+	// ClusterPropagationPolicy already exists, not to create it
+	return "", fmt.Errorf("the ClusterPropagationPolicy(%s) already exist, please edit it to propagate resource", policyName)
+}
+
+// preprocessResource delete redundant fields to convert resource as template
+func preprocessResource(obj *unstructured.Unstructured) error {
+	// remove fields that generated by kube-apiserver and no need(or can't) propagate to member clusters.
+	if err := removeIrrelevantField(obj); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// removeIrrelevantField used to remove fields that generated by kube-apiserver and no need(or can't) propagate to
+// member clusters.
+func removeIrrelevantField(workload *unstructured.Unstructured, extraHooks ...func(*unstructured.Unstructured)) error {
+	// populated by the kubernetes.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "creationTimestamp")
+
+	// populated by the kubernetes.
+	// The kubernetes will set this fields in case of graceful deletion. This field is read-only and can't propagate to
+	// member clusters.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "deletionTimestamp")
+
+	// populated by the kubernetes.
+	// The kubernetes will set this fields in case of graceful deletion. This field is read-only and can't propagate to
+	// member clusters.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "deletionGracePeriodSeconds")
+
+	// populated by the kubernetes.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "generation")
+
+	// This is mostly for internal housekeeping, and users typically shouldn't need to set or understand this field.
+	// Remove this field to keep 'Work' clean and tidy.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "managedFields")
+
+	// populated by the kubernetes.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "resourceVersion")
+
+	// populated by the kubernetes and has been deprecated by kubernetes.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "selfLink")
+
+	// populated by the kubernetes.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "uid")
+
+	unstructured.RemoveNestedField(workload.Object, "metadata", "ownerReferences")
+
+	unstructured.RemoveNestedField(workload.Object, "metadata", "finalizers")
+
+	unstructured.RemoveNestedField(workload.Object, "status")
+
+	if workload.GetKind() == common.ServiceKind {
```

Comment:
Pod maybe be deleted if it specifies not exist node name by k8s, should we remove spec.NodeName when Kind is Pod? cc @mrlihanbo